src/Controllers/User/OrderController.php

```diff
@@ -283,6 +284,71 @@ public function process(ServerRequest $request, Response $response, array $args)
         ]);
     }
 
+    public function processTopupOrder(ServerRequest $request, Response $response, array $args): ResponseInterface
+    {
+        $amount = $this->antiXss->xss_clean($request->getParam('amount'));
+
+        $amount = is_numeric($amount) ? (float) $amount : null;
+        if ($amount === null || $amount <= 0 || Tools::getDecimalPlaces($amount) > 2) {
```

Comment:
Should just use round(), and all pricing data in SSP are using double as data type, even if one long float value slips through, the value will still be truncated by the database.

Reply:
Updated.